### PR TITLE
d1: raw() vs all() vs run() fixes

### PR DESCRIPTION
**Pull request by elithrar**
This PR:

- [x] Adds a changelog for the breaking change (and fix) made to `raw()`
- [x] Updates the /query-databases/ docs to better describe `raw()` vs. `all()` behaviour
- [x] Documents how to use generic type params
- [x] Fixes docs for `first()`

### CODE REVIEW

File: content/d1/build-databases/query-databases.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ D1 automatically converts supported JavaScript (including TypeScript) types pass
 
 ## Return object
 
-The methods `stmt.run()`, `stmt.all()` and `db.batch()` return a typed `D1Result` object that contains the results (if applicable), the success status, and a meta object with the internal duration of the operation in milliseconds.
+The methods `stmt.raw()`, `stmt.all()` and `db.batch()` return a typed `D1Result` object that contains the results (if applicable), the success status, and a meta object with the internal duration of the operation in milliseconds.
```
Comment (geelen) [inline content/d1/build-databases/query-databases.md]: `.raw()` returns just a bare array, so no metadata

File: content/d1/build-databases/query-databases.md [hunk 2]
```diff
@@ -187,7 +162,30 @@ Dave,29
 */
 ```
```
Comment (geelen) [inline content/d1/build-databases/query-databases.md:169]: Probably worth adding a `### await stmt.raw({ columnNames: true })` heading with the contents:

```
As above, but the first row returned contains the names of the columns:

const stmt = db.prepare('SELECT name, age FROM users LIMIT 3');
const [columns, ...rows] = await stmt.raw({columnNames: true});

console.log(columns); // ['name','age']
...
```

File: content/d1/build-databases/query-databases.md (hunk 1 above)
Comment (elithrar) [reply content/d1/build-databases/query-databases.md]: Fixed.

File: data/changelogs/d1.yaml [hunk 3]
```diff
@@ -5,9 +5,9 @@ entries:
   - publish_date: "2024-02-13"
     title: API changes to `.raw()`, `.all()` and `.run()`
     description: |-
-      D1's `.raw()`, `.all()` and `.run()` [query statement methods](/d1/build-databases/query-databases/#query-statement-methods) have been updated to reflect their intended behaviour and improving compatibility with ORMs.
+      D1's `.raw()`, `.all()` and `.run()` [query statement methods](/d1/build-databases/query-databases/#query-statement-methods) have been updated to reflect their intended behaviour and improve compatibility with ORMs libraries.
```
Comment (Maddy-Cloudflare) [inline data/changelogs/d1.yaml]:
```suggestion
      D1's `raw()`, `all()` and `run()` [query statement methods](/d1/build-databases/query-databases/#query-statement-methods) have been updated to reflect their intended behaviour and improve compatibility with ORMs libraries.
```

File: data/changelogs/d1.yaml [hunk 4]
```diff
@@ -5,9 +5,9 @@ entries:
...
       
-      `.raw()` now correctly returns results an array of arrays, allowing the correct handling of duplicate column names (such as when joining tables), as compared to `.all()`, which is unchanged and returns an array of objects. To include an array of column names in the results when using `.raw()`, use `.raw({columnNames: true})`.
+      `.raw()` now correctly returns results as an array of arrays, allowing the correct handling of duplicate column names (such as when joining tables), as compared to `.all()`, which is unchanged and returns an array of objects. To include an array of column names in the results when using `.raw()`, use `.raw({columnNames: true})`.
```
Comment (Maddy-Cloudflare) [inline data/changelogs/d1.yaml]:
```suggestion
      `raw()` now correctly returns results as an array of arrays, allowing the correct handling of duplicate column names (such as when joining tables), as compared to `all()`, which is unchanged and returns an array of objects. To include an array of column names in the results when using `raw()`, use `raw({columnNames: true})`.
```

**Review by Maddy-Cloudflare — APPROVED** [review]

**Pull request merged**